### PR TITLE
Updated Stylelint version

### PR DESCRIPTION
**Pull request by alexasselin008**
## Summary

There are multiple inconsistencies in the css files. This PR is a proposition of stylistic rules we could add to the sg-stylelint configs to standardise the writing of CSS

## What I did

- Update stylelint rules
- Fix files with broken rules

### CODE REVIEW

File: packages/components/src/button/src/Button.css [hunk 1]
```diff
@@ -351,12 +351,12 @@ a.o-ui-button {
 .o-ui-button-inherit-style.o-ui-button-primary .o-ui-button-end-icon {
     fill: var(--o-ui-white);
     color: var(--o-ui-white);
-    opacity: .9;
+    opacity: 0.9;
```
Comment (alexasselin008) [inline packages/components/src/button/src/Button.css:354]: @fraincs Any opinion on this? at the moment, we have both 0.X and .X. Should we settle on one. Do you prefer with or without the 0?

File: packages/components/src/card/src/Card.css [hunk 2]
```diff
@@ -118,7 +118,7 @@
     padding: var(--o-ui-sp-6);
 }
 
-.o-ui-card-horizontal > .o-ui-card-illustration+.o-ui-card-aside {
+.o-ui-card-horizontal > .o-ui-card-illustration + .o-ui-card-aside {
```
Comment (alexasselin008) [inline packages/components/src/card/src/Card.css:121]: @fraincs the spacing around these operator is inconsistent. Is this the way you would like to write them, or without the spaces?

File: .stylelintrc.js [hunk 3]
```diff
@@ -1,6 +1,43 @@
+/** @type {import('stylelint').Config} */
 module.exports = {
     extends: "@sharegate/stylelint-config-recommended",
     rules: {
         "linebreaks": "unix",
+
+        "color-hex-length": "long",
```
Comment (alexasselin008) [inline .stylelintrc.js:7]: Will be in generic config file when accepted

File: packages/components/src/button/src/Button.css (hunk 1 above)
Comment (fraincs) [reply packages/components/src/button/src/Button.css:354]: No strong opinion, although in the case of opacity it has no use, as the maximum is 1.0

File: packages/components/src/card/src/Card.css (hunk 2 above)
Comment (fraincs) [reply packages/components/src/card/src/Card.css:121]: This is what happens on save on my end, been fighting with them a lot ` + ` is perfect.

**Review by fraincs — CHANGES_REQUESTED** [review]
I would remove the trailing 0 rule as its not adding anything.

**Review by fraincs — APPROVED** [review]

**Pull request merged**